archivebox/extractors/headers.py

```diff
@@ -0,0 +1,74 @@
+__package__ = 'archivebox.extractors'
+
+from pathlib import Path
+
+from typing import Optional
+
+from ..index.schema import Link, ArchiveResult, ArchiveOutput
+from ..system import atomic_write
+from ..util import (
+    enforce_types,
+    get_headers,
+    is_static_file,
+)
+from ..config import (
+    TIMEOUT,
+    CURL_BINARY,
+    CURL_USER_AGENT,
+    CURL_VERSION,
+    CHECK_SSL_VALIDITY,
+)
+from ..logging_util import TimedProgress
+
+@enforce_types
+def should_save_headers(link: Link, out_dir: Optional[str]=None) -> bool:
+    out_dir = out_dir or link.link_dir
+    if is_static_file(link.url):
+        return False
+
+    output = Path(out_dir or link.link_dir) / 'headers.json'
+    return not output.exists()
+
+
+@enforce_types
+def save_headers(link: Link, out_dir: Optional[str]=None, timeout: int=TIMEOUT) -> ArchiveResult:
+    """Download site headers"""
+
+    out_dir = Path(out_dir or link.link_dir)
+    output_folder = out_dir.absolute()
+    output: ArchiveOutput = 'headers.json'
+
+    status = 'succeeded'
+    timer = TimedProgress(timeout, prefix='      ')
+
+    cmd = [
+        CURL_BINARY,
+        '--silent',
+        '--max-time', str(timeout),
+        '--location',
+        '--compressed',
+        '--head',
+        *(['--user-agent', '{}'.format(CURL_USER_AGENT)] if CURL_USER_AGENT else []),
+        *([] if CHECK_SSL_VALIDITY else ['--insecure']),
+        link.url,
+    ]
+    try:
+        json_headers = get_headers(link.url)
```

Comment:
@pirate this is the only detail about this. It is using requests internally to get the headers. Is this ok? (The error `cmd` is falling back to a `curl` command. Or do you prefer to use `curl` for that too?